### ISSUE TITLE
Possible to Separate Learning Rates for Text-Encoder in SDXL LoRA Model Training?

### ISSUE THREAD

**Issue by Gtonero** [kohya-ss/sd-scripts#1748]
Is it possible to set the lora [sdxl_train_network.py] text encoder learning rate separately?
similar to Flux, which is supported in sdxl_train.py
in Clip_L, there are fewer parameters than in Clip_G, making the text-encoder in Clip_L more prone to overfitting compared to Clip_G

![image](https://github.com/user-attachments/assets/d6d02972-3eb8-4ee5-afcd-5579c50d87ff)
### sdxl_train.py
```
    parser.add_argument(
        "--learning_rate_te1",
        type=float,
        default=None,
        help="learning rate for text encoder 1 (ViT-L) / text encoder 1 (ViT-L)の学習率",
    )
    parser.add_argument(
        "--learning_rate_te2",
        type=float,
        default=None,
        help="learning rate for text encoder 2 (BiG-G) / text encoder 2 (BiG-G)の学習率",
```

**Comment by Gtonero** [comment]
@kohya-ss I encountered the problem SDXL Text-encoder 2 layer 31 
which has a higher chance of overfitting than other layers

```
lora_te2_layers_0 average weight: 0.0068537355400622, max weight: 0.0383300781250000, parameters: 184,326
lora_te2_layers_1 average weight: 0.0068535194732249, max weight: 0.0407714843750000, parameters: 2,027,586
lora_te2_layers_2 average weight: 0.0067848078906536, max weight: 0.0415039062500000, parameters: 2,027,586
lora_te2_layers_3 average weight: 0.0074247149750590, max weight: 0.0437011718750000, parameters: 552,978
lora_te2_layers_4 average weight: 0.0069831511937082, max weight: 0.0383300781250000, parameters: 184,326
lora_te2_layers_5 average weight: 0.0068072001449764, max weight: 0.0373535156250000, parameters: 184,326
lora_te2_layers_6 average weight: 0.0067534586414695, max weight: 0.0378417968750000, parameters: 184,326
lora_te2_layers_7 average weight: 0.0066636074334383, max weight: 0.0383300781250000, parameters: 184,326
lora_te2_layers_8 average weight: 0.0067557776346803, max weight: 0.0363769531250000, parameters: 184,326
lora_te2_layers_9 average weight: 0.0067224819213152, max weight: 0.0351562500000000, parameters: 184,326
lora_te2_layers_10 average weight: 0.0068326829932630, max weight: 0.0380859375000000, parameters: 184,326
lora_te2_layers_11 average weight: 0.0068012732081115, max weight: 0.0390625000000000, parameters: 184,326
lora_te2_layers_12 average weight: 0.0069304942153394, max weight: 0.0371093750000000, parameters: 184,326
lora_te2_layers_13 average weight: 0.0068560522049665, max weight: 0.0378417968750000, parameters: 184,326
lora_te2_layers_14 average weight: 0.0068084658123553, max weight: 0.0407714843750000, parameters: 184,326
lora_te2_layers_15 average weight: 0.0068167415447533, max weight: 0.0388183593750000, parameters: 184,326
lora_te2_layers_16 average weight: 0.0068856854923069, max weight: 0.0402832031250000, parameters: 184,326
lora_te2_layers_17 average weight: 0.0068622482940555, max weight: 0.0368652343750000, parameters: 184,326
lora_te2_layers_18 average weight: 0.0068633882328868, max weight: 0.0380859375000000, parameters: 184,326
lora_te2_layers_19 average weight: 0.0068264589644969, max weight: 0.0375976562500000, parameters: 184,326
lora_te2_layers_20 average weight: 0.0067679784260690, max weight: 0.0390625000000000, parameters: 184,326
lora_te2_layers_21 average weight: 0.0068171336315572, max weight: 0.0366210937500000, parameters: 184,326
lora_te2_layers_22 average weight: 0.0068559329956770, max weight: 0.0388183593750000, parameters: 184,326
lora_te2_layers_23 average weight: 0.0068042138591409, max weight: 0.0371093750000000, parameters: 184,326
lora_te2_layers_24 average weight: 0.0067991898395121, max weight: 0.0383300781250000, parameters: 184,326
lora_te2_layers_25 average weight: 0.0068012722767889, max weight: 0.0371093750000000, parameters: 184,326
lora_te2_layers_26 average weight: 0.0068148816935718, max weight: 0.0385742187500000, parameters: 184,326
lora_te2_layers_27 average weight: 0.0067317611537874, max weight: 0.0371093750000000, parameters: 184,326
lora_te2_layers_28 average weight: 0.0066674933768809, max weight: 0.0373535156250000, parameters: 184,326
lora_te2_layers_29 average weight: 0.0066365571692586, max weight: 0.0358886718750000, parameters: 184,326
lora_te2_layers_30 average weight: 0.0065232464112341, max weight: 0.0375976562500000, parameters: 184,326
lora_te2_layers_31 average weight: 0.0087950695306063, max weight: 0.0437011718750000, parameters: 184,326
```

Weight of Layer 31 is higher than the other layers even though the parameters are no different from the other layers

I experimented many times and found that when adding TE-lr the text-encoder layer 31 weight was too much.
```
lora_te2_layers_26 average weight: 0.0069863637909293, max weight: 0.0380859375000000, parameters: 92,166
lora_te2_layers_27 average weight: 0.0068571250885725, max weight: 0.0368652343750000, parameters: 92,166
lora_te2_layers_28 average weight: 0.0068515087477863, max weight: 0.0375976562500000, parameters: 92,166
lora_te2_layers_29 average weight: 0.0068382024765015, max weight: 0.0368652343750000, parameters: 92,166
lora_te2_layers_30 average weight: 0.0068043973296881, max weight: 0.0351562500000000, parameters: 92,166
lora_te2_layers_31 average weight: 0.0106498682871461, max weight: 0.0451660156250000, parameters: 92,166
```
causes a prompt or trigger word to cause unwanted things to enter the image.
If I set TE-lr low, the text-encoder 1 [SD1.5] doesn't learn well enough and the image will lack detail.
